fps-analysis/run-all-traces.py

```diff
@@ -0,0 +1,198 @@
+    if os.path.exists(results_directory):
+        shutil.rmtree(results_directory)
+    os.mkdir(results_directory)
+
+    if args.disable_cache_run is False:
+        #FIXME: add a verbose mode?
+        print("Warming up shader cache")
+        run_traces(args, traces_directory, None, 1)
+
+    fps_file = open(results_directory + '/fps-stats.txt', 'w')
+
+    #FIXME: add a verbose mode?
+    print("Starting FPS run")
+    run_traces(args, traces_directory, fps_file, args.num_samples)
+    fps_file.close()
+
+def main():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("--rebind", nargs='?', default=False, type=bool, help="If we call gfxrecon-replay with -m rebind")
+    parser.add_argument("--disable-cache-run", nargs='?', default=False, type=bool, help="By default we do a first run to ensure a hot shader cache, not included for the fps stats")
+    parser.add_argument("--traces-directory", nargs='?', default="traces", type=str, help="Directory with the traces")
+    parser.add_argument("--num-samples", nargs='?', default=1, type=int, help="Number of times each trace is executed to get the (averaged) fps value. Not include the shaderdb run")
+    parser.add_argument("--skip-gfxrecon", action="store_true", help="If we should skip running the gfxreconstruct traces")
+    parser.add_argument("--skip-apitrace", action="store_true", help="If we should skip running the apitrace traces")
+    parser.add_argument("--mesa-directory", nargs='?', type=str, help="Mesa directory. Useful if we pass a list of mesa commits")
+    parser.add_argument("--mesa-commit-list", nargs='+', action="extend", type=str, help="List of mesa commits to execute the script against")
+
+    args = parser.parse_args()
+
+    if args.mesa_commit_list is not None and len(args.mesa_commit_list) > 10:
+        print("List of mesa commits too big (maximum value is 10)")
+        return
+
+    if args.skip_gfxrecon and args.skip_apitrace:
+        print("Both --skip-gfxrecon and --skip-apitrace options used. Nothing to do")
+        return
+
+    traces_directory = os.path.expanduser(args.traces_directory)
+    print(traces_directory)
+
+    # We ensure that the on-disk-cache is enabled, as we want hot-cache fps numbers
+    os.unsetenv("MESA_GLSL_CACHE_DISABLE")
+    os.unsetenv("MESA_SHADER_CACHE_DISABLE")
+
+    # FIXME: perhaps further configuration on the results folder
+    # through a command line option?
+    base_results_directory = "results"
+
+    if (args.mesa_commit_list is None):
+        run_helper(args, traces_directory, base_results_directory)
+    else:
+        index = 0
+        for commit in args.mesa_commit_list:
+            results_directory = base_results_directory + "-" + str(index) + "-" + commit
+            mesa_directory = os.path.expanduser(args.mesa_directory)
+            command =  ['git', 'checkout', commit]
+            print(command)
+
+            try:
+                subprocess.run(command, cwd=mesa_directory)
```

Comment:
```suggestion
                subprocess.run(command, cwd=mesa_directory, check=True)
```
otherwise you won't get the exception when the command fails

(same for the other two `.run()` below)

Reply:
done.